source/common/router/header_formatter.h

```diff
@@ -25,6 +25,12 @@ class HeaderFormatter {
    *              headers
    */
   virtual bool append() const PURE;
+
+  /**
+   * @return bool indicating whether the formatted header should be skipped if the existing header
+   * is already set.
+   */
+  virtual bool skipIfPresent() const PURE;
```

Comment:
as we add more functions to this interface that are identical in all implementations it might be better to adjust the inheritance to declare the member variables, and implement the shared functions just once. Wdyt?

Reply:
that sounds great to me. It seemed to me that the codebase conventionally has a lot of "pure abstract" classes so I was hesitant to deviate, but there is a lot of duplication here that can just be solved in the base class.

Reply:
Done. I _could_ do the same for `append`, but figured that'd be out-of-scope for this change and can do so in a follow-up.

Reply:
looks good. I would just go ahead and move it here. But up to you! Thanks for cleaning it up :)

Reply:
done!